util-tracking/src/test/scala/sbt/util/TrackedSpec.scala

```diff
@@ -113,6 +135,27 @@ class TrackedSpec extends FlatSpec {
     }
   }
 
+  "outputChangedWriter" should "not require the input to have a JsonReader instance" in {
+    case class Input(v: Int)
+
+    implicit val writer = new JsonWriter[Input] {
+      override def write[J](obj: Input, builder: Builder[J]): Unit = builder.writeInt(obj.v)
+    }
+
+    withStore { store =>
+      val input0 = Input(1)
+
+      val cachedFun = Tracked.outputChangedWriter[Input, Int](store) {
```

Comment:
I am not very satisfied with the suffix but I didn't manage to make a true overload as the signature was ambiguous with the existing one. I did manage to create an overload by adding a dummy type parameter, but since the type parameters are usually explicitly passed (to avoid typing the arguments of the closure), it forced the caller to add that third, useless type.

Reply:
(and the naive idea of simply changing the type caused binary incompatibilty)
```
[error]  * method outputChanged(java.io.File,scala.Function2,sjsonnew.JsonFormat)scala.Function1 in object sbt.util.Tracked in current version does not have a correspondent with same parameter signature among (java.io.File,scala.Function2,sjsonnew.JsonWriter)scala.Function1, (sbt.util.CacheStore,scala.Function2,sjsonnew.JsonWriter)scala.Function1
```

Reply:
Assuming most plugin authors can continue to use the old variant, what do you think about using `W` as suffix - see https://github.com/sbt/sbt/pull/5513/files#r414623727 ?